filepaths/steampipe.go

```diff
@@ -19,53 +19,58 @@ const (
 
 var SteampipeDir string
 
-func steampipeSubDir(dirName string) string {
+func steampipeSubDir(dirName string, createOnNotExist bool) string {
 	if SteampipeDir == "" {
 		panic(fmt.Errorf("cannot call any Steampipe directory functions before SteampipeDir is set"))
 	}
 	subDir := filepath.Join(SteampipeDir, dirName)
 
-	if _, err := os.Stat(subDir); os.IsNotExist(err) {
+	if _, err := os.Stat(subDir); os.IsNotExist(err) && createOnNotExist {
 		err = os.MkdirAll(subDir, 0755)
 		utils.FailOnErrorWithMessage(err, fmt.Sprintf("could not create %s directory", dirName))
 	}
 
 	return subDir
 }
 
+// TmpDir returns the path to the tmp directory in STEAMPIPE_HOME (creates if missing)
```

Comment:
OK lets rename all the dir functions `Ensurexxxxx`, e.,g. `EnsureTemplateDir`

It's a bit unexpected that calling what seems like a simple name function creates the directory - I've never quite liked that

Then we can have the functions `TmpDir` and `EnsureTmpDir`